packages/pieces/apps/src/lib/generatebanners/actions/renderTemplate.action.ts

```diff
@@ -0,0 +1,191 @@
+import {
+  createAction,
+  Property,
+  AuthenticationType,
+  HttpRequest,
+  HttpMethod,
+  httpClient,
+  BasicAuthPropertyValue,
+} from '@activepieces/framework';
+
+export const renderTemplate = createAction({
+  name: 'render_template',
+  description: 'Render a GenerateBanners template',
+  displayName: 'Render Template',
+  props: {
+    authentication: Property.BasicAuth({
+      displayName: 'API Key',
+      description: 'Find your api key in your GenerateBanners account page.',
+      required: true,
+      username: {
+        displayName: 'Public API Key',
+      },
+      password: {
+        displayName: 'Secret API Key',
+      },
+    }),
+    template_id: Property.Dropdown({
+      displayName: 'Template',
+      required: true,
+      refreshers: ['authentication'],
+      options: async ({ authentication }) => {
+        const auth = authentication as BasicAuthPropertyValue;
+        if (!auth) {
+          return {
+            disabled: true,
+            options: [],
+            placeholder: 'Please enter your API key first.',
+          };
+        }
+
+        const response =
+          await httpClient.sendRequest<GenerateBannersTemplateList>({
+            method: HttpMethod.GET,
+            url: `https://api.generatebanners.com/api/v1/${auth['username']}/template`,
+            authentication: {
+              type: AuthenticationType.BEARER_TOKEN,
+              token: auth['password'],
+            },
+          });
+
+        if (response.status === 200) {
+          return {
+            disabled: false,
+            options: response.body.templates.map((template) => {
+              return {
+                label: template.name,
+                value: template.id,
+              };
+            }),
+          };
+        }
+
+        return {
+          disabled: true,
+          options: [],
+          placeholder: 'Error fetching templates',
+        };
+      },
+    }),
+    filetype: Property.StaticDropdown({
+      displayName: 'File type',
+      required: true,
+      defaultValue: 'jpg',
+      options: {
+        options: [
+          {
+            label: 'Image (jpg)',
+            value: 'jpg',
+          },
+          {
+            label: 'Image (png)',
+            value: 'png',
+          },
+          {
+            label: 'Document (pdf)',
+            value: 'pdf',
+          },
+          {
+            label: 'Video (mp4)',
+            value: 'mp4',
+          },
+        ],
+      },
+    }),
+    audio_url: Property.ShortText({
+      displayName: 'Audio url',
+      description: 'Link to an audio file. Only used for videos.',
+      required: false,
+    }),
+    variables: Property.DynamicProperties({
+      displayName: 'Variables',
+      required: false,
```

Comment:
I think this should be required for better user experience, what do you think?

Reply:
I was confused, I thought "required" meant the user had to fill all the inputs shown.
After testing, I agree it's better to always show these fields instead of hiding them in the optional section 👌

I've updated it to required. Good catch, thanks!